Use psammead-image in Simorgh

radiocontrolled opened the pull request:
Resolves https://github.com/BBC-News/simorgh/issues/1077

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

radiocontrolled committed:
add  psammead image package

radiocontrolled committed:
remove image component

radiocontrolled committed:
update Image and Figure containers

sareh commented on src/app/containers/Figure/index.jsx:
```diff
@@ -11,10 +11,18 @@ const renderCopyright = copyright =>
   <Figure>
     <ImagePlaceholder ratio={ratio}>
-      <Image alt={alt} src={src} />
+      <Image alt={alt} src={src} width={width} height={height} />
```
```suggestion
      <Image alt={alt} src={src} width={width} />
```
This is because adding the height too makes the image stretch:
With `height` attribute:
<img width="511" alt="screen shot 2018-12-19 at 16 38 42" src="https://user-images.githubusercontent.com/3028997/50234026-a5a7b300-03ac-11e9-8281-23f29184b238.png">
Without `height` attribute:
<img width="506" alt="screen shot 2018-12-19 at 16 41 03" src="https://user-images.githubusercontent.com/3028997/50234163-f1f2f300-03ac-11e9-802a-8400d8555407.png">

sareh commented on src/app/containers/Figure/index.jsx:
```diff
@@ -23,10 +31,12 @@ const FigureContainer = ({ src, alt, ratio, copyright, captionBlock }) => (
-  copyright: string,
   captionBlock: objectOf(any),
+  copyright: string,
+  height: number.isRequired,
```
```suggestion
```

radiocontrolled replied:
I have applied your comments @sareh

sareh commented on src/app/containers/Figure/index.jsx:
```diff
@@ -11,10 +11,18 @@ const renderCopyright = copyright =>
+  alt,
+  copyright,
+  captionBlock,
+  height,
```
```suggestion
```

sareh reviewed (changes requested):
Looks good so far. I've just added comments about how to ensure the image doesn't stretch.

sareh committed:
Update src/app/containers/Figure/index.jsx

Co-Authored-By: radiocontrolled <radiocontrolled@gmail.com>

sareh committed:
Update src/app/containers/Figure/index.jsx

Co-Authored-By: radiocontrolled <radiocontrolled@gmail.com>

sareh committed:
Update src/app/containers/Figure/index.jsx

Co-Authored-By: radiocontrolled <radiocontrolled@gmail.com>

bcmn reviewed (commented):
Looks good overall -- is this error just because Psammead hasn't got similar stretching fixes yet?
![screenshot 2018-12-20 at 11 23 07](https://user-images.githubusercontent.com/11646957/50282151-bbb78100-0449-11e9-8408-88757fc24184.png)

ChrisBAshton commented:
we should remove the `.required` on `height`

sareh commented on package.json:
```diff
@@ -62,6 +62,7 @@
     "@bbc/psammead-caption": "^0.1.0",
     "@bbc/psammead-copyright": "^0.2.0",
     "@bbc/psammead-headings": "^0.1.2",
+    "@bbc/psammead-image": "^0.1.0",
```
If you pull in `^0.2.0`, the prop error in the console should disappear. (Changes to package-lock need to be made, too)
```suggestion
    "@bbc/psammead-image": "^0.2.0",
```

radiocontrolled committed:
updates version of psammead-image

sareh commented on src/app/containers/Figure/__snapshots__/index.test.jsx.snap:
```diff
@@ -34,7 +34,9 @@ exports[`Figure should render an image with alt text 1`] = `
     <img
       alt="Pauline Clayton"
       className="c2"
+      height={360}
```
The snapshots need to be updated.

radiocontrolled committed:
update snap

radiocontrolled commented:
@ChrisBAshton there isn't a required on height
@bcmn errors do not appear with the newer version of `@bbc/psammead-image`
@sareh I have addressed your comments

sareh reviewed (approved):
📸 Looks great!

jamesbrumpton commented:
LGTM 👍
